### PR TITLE
some editor bugfixes from #8903

### PR DESCRIPTION
**Pull request by babaissarkar**
Contains a couple of backported fixes from #8903, and some new behavioral fixes inside the Editor. Changes detailed in the commit messages. New changes will need to be forward-ported.

New commits in this PR (forward-port needed):
4b1752be06b396abf3201b9fc4e14beccd2127ab

### CODE REVIEW

File: src/editor/map/context_manager.cpp [hunk 1]
```diff
@@ -675,9 +675,11 @@ void context_manager::resize_map_dialog()
 
 void context_manager::save_map_as_dialog()
 {
+	bool first_pick = false;
 	std::string input_name = get_map_context().get_filename();
 	if(input_name.empty()) {
-		input_name = filesystem::get_legacy_editor_dir()+"/maps";
+		first_pick = true;
+		input_name = filesystem::get_current_editor_dir(editor_controller::current_addon_id_)+"/maps";
```
Comment (soliton-) [inline src/editor/map/context_manager.cpp:682]: What happens if there is no addon selected?

Comment (babaissarkar) [reply src/editor/map/context_manager.cpp:682]: I missed implementing that part, any idea what should it do?

Comment (babaissarkar) [reply src/editor/map/context_manager.cpp:682]: probably will need to be forward-ported.

Comment (babaissarkar) [reply src/editor/map/context_manager.cpp:682]: will save to the editor/maps directory.

Comment (soliton-) [reply src/editor/map/context_manager.cpp:682]: Looks like the dialog goes to `<userdata>/data/addons` then (that's what happens in master anyway). It should go to `<userdata>/editor/maps` as before.

Comment (babaissarkar) [reply src/editor/map/context_manager.cpp]: Fixed in 675e7620e67fa5df1ac09ae784c1fc070b128d99

**Comment by babaissarkar** [comment]
675e7620e67fa5df1ac09ae784c1fc070b128d99 is a new commit in this PR, and might need to be forward-ported.

File: src/editor/controller/editor_controller.cpp [hunk 2]
```diff
@@ -876,6 +876,14 @@ bool editor_controller::do_execute_command(const hotkey::ui_command& cmd, bool p
 			un->anim_comp().set_standing();
 		}
 		return true;
+		case HOTKEY_EDITOR_UNIT_TOGGLE_LOYAL:
+		{
+			map_location loc = gui_->mouseover_hex();
+			const unit_map::unit_iterator un = get_current_map_context().units().find(loc);
+			bool loyal = un->loyal();
```
Comment (soliton-) [inline src/editor/controller/editor_controller.cpp:883]: What happens if no unit is found?

Comment (babaissarkar) [reply src/editor/controller/editor_controller.cpp:883]: The "Loyal" checkbox is not visible unless there is actually an unit on that hex (so this cannot be triggered). I'll try to add a failsafe.

Comment (soliton-) [reply src/editor/controller/editor_controller.cpp:883]: > The "Loyal" checkbox is not visible unless there is actually an unit on that hex (so this cannot be triggered).

You can bind the hotkey to something and trigger it with any hex selected.

Comment (babaissarkar) [reply src/editor/controller/editor_controller.cpp:883]: Still doesn't cause any problems (no crash or something).

Comment (babaissarkar) [reply src/editor/controller/editor_controller.cpp:883]: One thing I noticed though is that when that action is triggered by hotkey, the unit didn't redraw until I clicked somewhere else.

Comment (soliton-) [reply src/editor/controller/editor_controller.cpp:883]: If this is defined behavior then it's all good. If it's undefined behavior then not crashing means nothing.

**Pull request merged**